### PR TITLE
Week1setup 

### PR DESCRIPTION
**Pull request by EugeneTeu**
https://eugene-sps.uc.r.appspot.com/

### CODE REVIEW

**Comment by EugeneTeu** [comment]
hi @fanwgwg , sorry for the lengthy pr. 
portfolio is up at https://eugene-sps.uc.r.appspot.com/

ready for review thanks!

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -1,18 +1,276 @@
 <!DOCTYPE html>
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/index.html:1]: Please keep the original indentation here (2 space instead tab), for better readability and review

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -12,17 +12,23 @@
...
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function openSection(evt, sectionName) {
+  // Declare all variables
+  var i, tabcontent, tablinks;
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/script.js:17]: A few things here:
1. These comments aren't meaningful comments
2. I don't see a universal naming convention being applied here, the param name is `sectionName`, while here it's `tabcontent`, `tablinks` e.g, it should be tabContent, tabLinks? 
3. Why is `i` needed to be declared? Is it more error-prone to create a shared variable for indexing between different arrays?
4. tabContent seems to be a list/array, its name isn't reflecting that, while `tabLinks` is more obvious

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -1,9 +1,240 @@
+:root {
+    overflow-x: auto;
+     font-size: 16px;
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/style.css:3]: indentation?

File: portfolio/src/main/webapp/style.css [hunk 4]
```diff
@@ -1,9 +1,240 @@
...
+    margin: 0;
+    padding: 0;
+}
+body::-webkit-scrollbar {
+    width: 0.25rem;;
+}
+body::-webkit-scrollbar-track{
+    background: black;
+}
+
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/style.css:25]: nit: Remove the extra space or keep one space between all the sections

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -12,17 +12,23 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function openSection(evt, sectionName) {
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/script.js:15]: Add doc to public methods

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -12,17 +12,23 @@
...
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  // Get all elements with class="tablinks" and remove the class "active"
+  tablinks = document.getElementsByClassName("tablinks");
+  for (i = 0; i < tablinks.length; i++) {
+    tablinks[i].className = tablinks[i].className.replace(" active", "");
+  }
+
+  // Show the current tab, and add an "active" class to the button that opened the tab
+  document.getElementById(sectionName).style.display = "block";
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/script.js:32]: Not sure if there's ever a possibility that they could be null, however, shall we add a null check for `document.getElementById(sectionName)`

File: portfolio/src/main/webapp/index.html [hunk 7]
```diff
@@ -1,18 +1,276 @@
...
+	</div>
+	<script>
+		$(function(){
+  $("#nav-placeholder").load("nav.html");
+});
+
+	</script>
+	<!--end of Navigation bar-->
+	<div id="content">
+		<div id="Home" class="tabcontent">
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/index.html:29]: Please use lowercase for all HTML element names, attributes, attribute values (unless text), CSS selectors, properties, and property values (with the exception of strings).

https://google.github.io/styleguide/htmlcssguide.html#Capitalization

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (EugeneTeu) [reply portfolio/src/main/webapp/index.html:1]: @fanwgwg  hi weiguang, i have an issue where git does not track the indentation for my HTML code. I converted it back to format to spaces (with setting of 2) using the IDE and it looks fine on my google cloud shell: 

image

but somehow, I am unable to commit this diff to git. Is there a workaround for this?

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -12,17 +12,23 @@
...
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+/*
+* Finds the tab that has been selected and display that tab
```
Comment (EugeneTeu) [inline portfolio/src/main/webapp/script.js:16]: @fanwgwg  Not sure if this public doc is comprehensive enough? LMK!

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (EugeneTeu) [reply portfolio/src/main/webapp/script.js:21]: I added one! LMK if it is not up comprehensive enough

File: portfolio/src/main/webapp/style.css (hunk 3 above)
Comment (EugeneTeu) [reply portfolio/src/main/webapp/style.css]: fixed!

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (EugeneTeu) [reply portfolio/src/main/webapp/script.js]: added null check

File: portfolio/src/main/webapp/index.html [hunk 9]
```diff
@@ -1,18 +1,264 @@
...
+  <!--end of Navigation bar-->
+  <div id="content">
+    <div id="home" class="tab-content">
+      <h1 class="m-title">Eugene Teu</h1>
+      <h2>Software Engineering. Computer Science.</h2>
     </div>
-  </body>
-</html>
+    <!-- Tab content -->
+    <div id="About" class="tab-content">
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/index.html:28]: Thought we agreed that these should be lowercased?

Comment (fanwgwg) [reply portfolio/src/main/webapp/index.html:28]: Same applies to other ids in the following up tab-content(s)

File: portfolio/src/main/webapp/index.html [hunk 10]
```diff
@@ -1,18 +1,264 @@
...
+
+      <div id="Skills" class="tab-content">
+        <div>
+          <h2>Skills</h2>
+          <div class="job">
+            <dl class="job-item">
+              <dt>Languages</dt>
+              <ul>
+                <li>
+                  Java
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/index.html:60]: Prefer `<li>...</li>` over new lines everywhere for readability

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (fanwgwg) [reply portfolio/src/main/webapp/script.js:21]: Feel free to resolve comments if you believe that they were resolved :)

File: portfolio/src/main/webapp/script.js [hunk 11]
```diff
@@ -12,17 +12,25 @@
...
+*
+* @params { object } evt: HTML event
+* @params { sectionName } string: section Nam
```
Comment (fanwgwg) [inline portfolio/src/main/webapp/script.js:19]: Left the `e` here for Name?

File: portfolio/src/main/webapp/index.html (hunk 9 above)
Comment (EugeneTeu) [reply portfolio/src/main/webapp/index.html]: sorry my bad, missed out the other ids

File: portfolio/src/main/webapp/index.html (hunk 10 above)
Comment (EugeneTeu) [reply portfolio/src/main/webapp/index.html]: ok , but for those that were too long i still gave break them to make it readable

File: portfolio/src/main/webapp/script.js (hunk 11 above)
Comment (EugeneTeu) [reply portfolio/src/main/webapp/script.js]: my bad

File: portfolio/src/main/webapp/index.html (hunk 10 above)
Comment (fanwgwg) [reply portfolio/src/main/webapp/index.html]: Yeap agreed, for long lines we should still add new lines to it

**Pull request merged**